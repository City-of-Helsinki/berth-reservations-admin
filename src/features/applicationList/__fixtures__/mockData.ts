import { BERTH_APPLICATIONS } from '../__generated__/BERTH_APPLICATIONS';
import { getBerthApplicationData } from '../utils';
import { ApplicationStatus, LeaseStatus } from '../../../@types/__generated__/globalTypes';

export const mockData: BERTH_APPLICATIONS = {
  berthApplications: {
    __typename: 'BerthApplicationNodeConnection',
    count: 3,
    edges: [
      // Full structure
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '1',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-0',
            pier: 'A',
            reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano',
          boatDraught: 5,
          boatRegistrationNumber: 'A 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Matti',
          harborChoices: [{ __typename: 'HarborChoiceType', harbor: '1', harborName: '1', priority: 0 }],
          id: 'MOCK-APPLICATION-0',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-0',
                    properties: { __typename: 'HarborProperties', name: 'Testisatama' },
                  },
                  identifier: 'G',
                },
              },
            },
            id: 'MOCK-LEASE-0',
            status: LeaseStatus.OFFERED,
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Most structure and nulled fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '2',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-1',
            pier: 'A',
            reason: null,
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano II',
          boatDraught: 5,
          boatRegistrationNumber: 'B 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Maija',
          harborChoices: null,
          id: 'MOCK-APPLICATION-1',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            id: 'MOCK-LEASE-1',
            status: LeaseStatus.OFFERED,
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  identifier: 'Pier Brosnan',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-1',
                    properties: {
                      __typename: 'HarborProperties',
                      name: null,
                    },
                  },
                },
              },
            },
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Minimum fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: null,
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano 3-D',
          boatDraught: 5,
          boatRegistrationNumber: 'C 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE-0' },
          firstName: 'Matti',
          harborChoices: [
            { __typename: 'HarborChoiceType', harbor: 'MOCK-HARBOR-0', harborName: 'Testisatama', priority: 0 },
          ],
          id: 'MOCK-APPLICATION-2',
          lastName: 'Möttönen',
          lease: null,
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
    ],
  },
  boatTypes: [{ __typename: 'BoatTypeType', id: '1', name: 'Purjevene' }],
};

export const mockTableData = getBerthApplicationData(mockData);
