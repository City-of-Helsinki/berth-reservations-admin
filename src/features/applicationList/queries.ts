import gql from 'graphql-tag';

export const BERTH_APPLICATIONS_QUERY = gql`
  query BERTH_APPLICATIONS(
    $first: Int!
    $after: String
    $switchApplications: Boolean
    $orderBy: String
    $statuses: [ApplicationStatus]
    $nameFilter: String
  ) {
    berthApplications(
      first: $first
      after: $after
      switchApplications: $switchApplications
      orderBy: $orderBy
      statuses: $statuses
      name: $nameFilter
    ) {
      count
      edges {
        node {
          id
          firstName
          lastName
          customer {
            id
          }
          berthSwitch {
            berthNumber
            harbor
            harborName
            id
            pier
            reason {
              title
            }
          }
          email
          createdAt
          municipality
          boatType
          boatRegistrationNumber
          boatWidth
          boatLength
          boatDraught
          boatWeight
          boatName
          boatModel
          accessibilityRequired
          status
          lease {
            id
            status
            order {
              id
            }
            berth {
              number
              pier {
                properties {
                  identifier
                  harbor {
                    id
                    properties {
                      name
                    }
                  }
                }
              }
            }
          }
          harborChoices {
            harbor
            priority
            harborName
          }
        }
      }
    }
    boatTypes {
      id
      name
    }
  }
`;
