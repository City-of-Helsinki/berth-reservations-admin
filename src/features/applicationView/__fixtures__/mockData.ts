import {
  INDIVIDUAL_APPLICATION,
  INDIVIDUAL_APPLICATION_additionalProducts as ADDITIONAL_PRODUCTS,
  INDIVIDUAL_APPLICATION_berthApplication as BERTH_APPLICATION,
  INDIVIDUAL_APPLICATION_berthApplication_berthSwitch as BERTH_SWITCH,
  INDIVIDUAL_APPLICATION_berthApplication_customer as CUSTOMER,
  INDIVIDUAL_APPLICATION_berthApplication_lease as LEASE,
  INDIVIDUAL_APPLICATION_boatTypes as BOAT_TYPES,
} from '../__generated__/INDIVIDUAL_APPLICATION';
import {
  AdditionalProductType,
  ApplicationStatus,
  BerthApplicationLanguage,
  BerthMooringType,
  CustomerGroup,
  InvoicingType,
  Language,
  PeriodType,
  PriceUnits,
  ProductServiceType,
} from '../../../@types/__generated__/globalTypes';

export const mockCustomer: CUSTOMER = {
  __typename: 'ProfileNode',
  comment: 'Comment',
  firstName: 'Testi',
  id: 'MOCK-CUSTOMER',
  invoicingType: InvoicingType.DIGITAL_INVOICE,
  language: Language.FINNISH,
  lastName: 'Testinen',
  organization: null,
  primaryAddress: { __typename: 'AddressNode', address: 'Testikatu 1', city: 'Helsinki', postalCode: '00100' },
  primaryEmail: { __typename: 'EmailNode', email: 'test@example.com' },
  primaryPhone: { __typename: 'PhoneNode', phone: '+358 00 000 0000' },
  customerGroup: CustomerGroup.PRIVATE,
};

export const mockBerthSwitch: BERTH_SWITCH = {
  __typename: 'BerthSwitchType',
  berthNumber: '1',
  harbor: 'MOCK-HARBOR',
  harborName: 'Test Harbor',
  id: 'MOCK-BERTH-SWITCH',
  pier: 'A',
  reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
};

export const mockLease: LEASE = {
  __typename: 'BerthLeaseNode',
  berth: {
    __typename: 'BerthNode',
    comment: '',
    depth: null,
    isAccessible: null,
    length: 5,
    mooringType: BerthMooringType.DINGHY_PLACE,
    number: 1,
    pier: {
      __typename: 'PierNode',
      properties: {
        __typename: 'PierProperties',
        electricity: false,
        gate: false,
        harbor: {
          __typename: 'HarborNode',
          id: 'MOCK-HARBOR',
          properties: { __typename: 'HarborProperties', name: 'Test Harbor' },
        },
        identifier: 'A',
        lighting: false,
        mooring: false,
        wasteCollection: false,
        water: false,
      },
    },
    width: 0,
  },
  id: 'MOCK-LEASE',
  order: null,
};

const additionalProducts: ADDITIONAL_PRODUCTS = {
  __typename: 'AdditionalProductNodeConnection',
  edges: [
    {
      __typename: 'AdditionalProductNodeEdge',
      node: {
        __typename: 'AdditionalProductNode',
        id: 'MOCK-ADDITIONAL-PRODUCT-NODE',
        period: PeriodType.MONTH,
        priceUnit: PriceUnits.AMOUNT,
        priceValue: '25.00',
        productType: AdditionalProductType.OPTIONAL_SERVICE,
        service: ProductServiceType.DINGHY_PLACE,
      },
    },
  ],
};

const berthApplication: BERTH_APPLICATION = {
  __typename: 'BerthApplicationNode',
  accessibilityRequired: false,
  address: 'Testikatu 1',
  berthSwitch: null,
  boatDraught: 1,
  boatLength: 1,
  boatModel: 'Marine',
  boatName: 'Cama la Yano',
  boatRegistrationNumber: '1234',
  boatType: '7',
  boatWeight: 120,
  boatWidth: 1,
  businessId: '',
  companyName: '',
  createdAt: '2020-07-10T08:04:36.062819+00:00',
  customer: null,
  email: 'test@example.com',
  firstName: 'Testi',
  harborChoices: [
    { harbor: 'MOCK-HARBOR-B', priority: 1, harborName: 'Saukonpaaden venesatama', __typename: 'HarborChoiceType' },
  ],
  id: 'MOCK-APPLICATION',
  language: BerthApplicationLanguage.FI,
  lastName: 'Testinen',
  lease: null,
  municipality: 'Helsinki',
  phoneNumber: '+358 00 000 0000',
  status: ApplicationStatus.PENDING,
  zipCode: '00100',
  acceptBoatingNewsletter: true,
  acceptFitnessNews: false,
  acceptLibraryNews: false,
  acceptOtherCultureNews: true,
  applicationCode: '1234',
};

const boatTypes: BOAT_TYPES[] = [
  { id: '1', name: 'Jollavene', __typename: 'BoatTypeType' },
  { id: '2', name: 'Soutuvene', __typename: 'BoatTypeType' },
  { id: '3', name: 'Perämoottorivene', __typename: 'BoatTypeType' },
  { id: '4', name: 'Sisäperämoottorivene', __typename: 'BoatTypeType' },
  { id: '5', name: 'Keskimoottorivene', __typename: 'BoatTypeType' },
  { id: '6', name: 'Purjevene / moottoripursi', __typename: 'BoatTypeType' },
  { id: '7', name: 'Troolari', __typename: 'BoatTypeType' },
  { id: '8', name: 'Suuri alus (yli 20t)', __typename: 'BoatTypeType' },
];

export const mockApplication: INDIVIDUAL_APPLICATION = {
  __typename: 'Query',
  additionalProducts: additionalProducts,
  berthApplication: berthApplication,
  boatTypes: boatTypes,
};
