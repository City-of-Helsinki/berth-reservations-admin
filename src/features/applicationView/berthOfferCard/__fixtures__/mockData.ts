import { BerthLease_lease as BerthLease } from '../__generated__/BerthLease';
import {
  AdditionalProductType,
  BerthMooringType,
  LeaseStatus,
  ProductServiceType,
} from '../../../../@types/__generated__/globalTypes';

export const lease: BerthLease = {
  id: '165023f5-5b74-4656-9315-d4e1a22f111d',
  status: LeaseStatus.DRAFTED,
  berth: {
    depth: null,
    length: 6,
    mooringType: BerthMooringType.SINGLE_SLIP_PLACE,
    width: 2.5,
    comment: '',
    isAccessible: null,
    number: '1',
    pier: {
      properties: {
        identifier: 'C',
        electricity: true,
        gate: true,
        lighting: true,
        mooring: true,
        wasteCollection: true,
        water: true,
        harbor: {
          id: 'b55da996-cf00-4b52-9aa0-6b13fd8d806b',
          properties: {
            name: 'Aurinkolahden venesatama (Aurinkoranta)',
            __typename: 'HarborProperties',
          },
          __typename: 'HarborNode',
        },
        __typename: 'PierProperties',
      },
      __typename: 'PierNode',
    },
    __typename: 'BerthNode',
  },
  customer: {
    __typename: 'ProfileNode',
    id: 'MOCK-CUSTOMER',
    primaryEmail: {
      __typename: 'EmailNode',
      id: 'MOCK-EMAIL',
      email: 'test@example.com',
    },
  },
  order: {
    id: '07f65f54-4a25-47fd-9eea-d9fdddfa46f9',
    orderNumber: '123456',
    price: '96.67',
    totalPrice: '256.70',
    orderLines: {
      edges: [
        {
          node: {
            id: '9b97b9bc-bba5-46d9-aae2-86e8686012ef',
            price: '20.50',
            product: {
              id: 'dc7f8158-a4cb-49a9-9368-e586f224fdb5',
              service: ProductServiceType.ELECTRICITY,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: 'ea2ef175-f790-45a0-b16f-9e5a98340691',
            price: '19.33',
            product: {
              id: '2e55f605-061e-4299-814d-bf2cd53aa8ce',
              service: ProductServiceType.WATER,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '8ba4de61-63cc-49fc-afe2-26068f3c8e4d',
            price: '2.42',
            product: {
              id: '3a2e3a2f-2782-40e9-a139-b9a4d4a36c3b',
              service: ProductServiceType.GATE,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '59c768c1-f371-4178-afda-1e16d2c725b0',
            price: '8.33',
            product: {
              id: '762da6e9-89be-43bc-b097-53dff4ed8415',
              service: ProductServiceType.MOORING,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '6d910d8e-80b6-4486-aa85-4958bcd98dd5',
            price: '4.17',
            product: {
              id: '34c5b6bc-7eb8-457e-bce8-8ba3a07b9057',
              service: ProductServiceType.WASTE_COLLECTION,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: 'ab7cf1e5-92b5-42ad-844a-e2d577d30baf',
            price: '40.28',
            product: {
              id: '650dbfcb-8f10-46b6-af64-197b77686a5b',
              service: ProductServiceType.LIGHTING,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '244a7af9-6f91-4f08-9738-82ef06a2e790',
            price: '65.00',
            product: {
              id: '5194dd14-0226-4603-808c-831cfbec72de',
              service: ProductServiceType.DINGHY_PLACE,
              productType: AdditionalProductType.OPTIONAL_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
      ],
      __typename: 'OrderLineNodeConnection',
    },
    __typename: 'OrderNode',
  },
  __typename: 'BerthLeaseNode',
};
