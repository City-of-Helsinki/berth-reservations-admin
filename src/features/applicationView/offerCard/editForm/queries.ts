import gql from 'graphql-tag';

export const ORDER_OPTIONAL_PRODUCTS_QUERY = gql`
  query ORDER_OPTIONAL_PRODUCTS {
    additionalProducts(productType: OPTIONAL_SERVICE) {
      edges {
        node {
          id
          service
          priceValue
          priceUnit
          period
          productType
        }
      }
    }
  }
`;
