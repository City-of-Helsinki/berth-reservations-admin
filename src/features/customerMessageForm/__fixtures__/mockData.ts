import { NOTIFICATION_TEMPLATES } from '../__generated__/NOTIFICATION_TEMPLATES';
import { NotificationTemplateLanguage } from '../../../@types/__generated__/globalTypes';

export const mockHtml =
  // eslint-disable-next-line max-len
  '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="fi" xml:lang="fi"><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta name="viewport" content="width=device-width"><title>Test HTML</title></head><body><h1>Test HTML</h1><p>Test HTML</p></body></html>';

export const mockData: NOTIFICATION_TEMPLATES = {
  notificationTemplates: {
    edges: [
      {
        node: {
          id: 'Tm90aWZpY2F0aW9uVGVtcGxhdGVOb2RlOjE=',
          preview: mockHtml,
          translations: [
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.EN,
              preview: mockHtml,
              subject: 'Confirmation: Berth application',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.SV,
              preview: mockHtml,
              subject: 'Bekräftelse: båtplatsansökning',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.FI,
              preview: mockHtml,
              subject: 'Vahvistus: venepaikan hakemus',
              __typename: 'NotificationTranslationType',
            },
          ],
          type: 'berth_application_created',
          __typename: 'NotificationTemplateNode',
        },
        __typename: 'NotificationTemplateNodeEdge',
      },
      {
        node: {
          id: 'Tm90aWZpY2F0aW9uVGVtcGxhdGVOb2RlOjI=',
          preview: mockHtml,
          translations: [
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.FI,
              preview: mockHtml,
              subject: 'Vahvistus: Veneen talvisäilytyspaikan hakemus',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.SV,
              preview: mockHtml,
              subject: 'Bekräftelse: ansökan om vinteruppläggningsplats',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.EN,
              preview: mockHtml,
              subject: 'Confirmation: boat winter storage application',
              __typename: 'NotificationTranslationType',
            },
          ],
          type: 'winter_storage_application_created',
          __typename: 'NotificationTemplateNode',
        },
        __typename: 'NotificationTemplateNodeEdge',
      },
    ],
    __typename: 'NotificationTemplateNodeConnection',
  },
};
