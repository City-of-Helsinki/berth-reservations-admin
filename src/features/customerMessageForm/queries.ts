import gql from 'graphql-tag';

export const NOTIFICATION_TEMPLATES_QUERY = gql`
  query NOTIFICATION_TEMPLATES {
    notificationTemplates {
      edges {
        node {
          id
          preview
          translations {
            bodyHtml
            bodyText
            languageCode
            preview
            subject
          }
          type
        }
      }
    }
  }
`;
