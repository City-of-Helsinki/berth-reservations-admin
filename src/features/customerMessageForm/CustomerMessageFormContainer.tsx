import React, { useState } from 'react';
import { useQuery } from '@apollo/react-hooks';

import { CustomerMessageForm } from './CustomerMessageForm';
import { NOTIFICATION_TEMPLATES_QUERY } from './queries';
import { NOTIFICATION_TEMPLATES } from './__generated__/NOTIFICATION_TEMPLATES';
import { getNotificationTemplates } from './utils';
import LoadingSpinner from '../../common/spinner/LoadingSpinner';
import { MessageFormValues } from './types';

export interface CustomerMessageFormContainerProps {
  closeModal: () => void;
  handleSendMessage: () => void;
  selectedCustomerIds: string[];
}

const CustomerMessageFormContainer = ({
  closeModal,
  handleSendMessage,
  selectedCustomerIds,
}: CustomerMessageFormContainerProps) => {
  const { data, loading } = useQuery<NOTIFICATION_TEMPLATES>(NOTIFICATION_TEMPLATES_QUERY);
  const [previewHtml, setPreviewHtml] = useState<string | undefined>(undefined);

  if (loading || !data?.notificationTemplates) return <LoadingSpinner />;

  const notificationTemplates = getNotificationTemplates(data);

  const templateOptions = notificationTemplates.map(({ id, type }) => {
    return {
      value: id,
      label: type,
    };
  });

  const getPreviewHtml = (templateId: string) =>
    notificationTemplates.find((template) => {
      return template.id === templateId;
    })?.preview || undefined;

  const handlePreview = (templateId: string) => {
    setPreviewHtml(getPreviewHtml(templateId));
  };

  const onSubmit = (message: MessageFormValues) => {
    alert(`CustomerIds: ${JSON.stringify(selectedCustomerIds)} content: ${JSON.stringify(message)}`);
    handleSendMessage();
  };

  return (
    <CustomerMessageForm
      handleCancel={closeModal}
      handleCancelPreview={() => setPreviewHtml(undefined)}
      handlePreview={handlePreview}
      handleSendMessage={onSubmit}
      previewHtml={previewHtml}
      recipientCount={selectedCustomerIds.length}
      templateOptions={templateOptions}
    />
  );
};

export default CustomerMessageFormContainer;
