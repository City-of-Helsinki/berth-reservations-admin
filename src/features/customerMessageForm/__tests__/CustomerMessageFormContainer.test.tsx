import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { MockedProvider } from '@apollo/react-testing';
import { act } from 'react-dom/test-utils';
import waitForExpect from 'wait-for-expect';
import CustomerMessageFormContainer, { CustomerMessageFormContainerProps } from '../CustomerMessageFormContainer';
import { mockData, mockHtml } from '../__fixtures__/mockData';
import { NOTIFICATION_TEMPLATES_QUERY } from '../queries';
import LoadingSpinner from '../../../common/spinner/LoadingSpinner';

const mockProps: CustomerMessageFormContainerProps = {
  closeModal: jest.fn(),
  handleSendMessage: jest.fn(),
  selectedCustomerIds: ['0', '1'],
};

const queryMock = {
  request: { query: NOTIFICATION_TEMPLATES_QUERY },
  result: {
    data: mockData,
  },
};

describe('CustomerMessageFormContainer', () => {
  const getWrapper = () =>
    mount(
      <MockedProvider mocks={[queryMock]}>
        <CustomerMessageFormContainer {...mockProps} />
      </MockedProvider>
    );

  const waitForContent = async (wrapper: ReactWrapper) => {
    await act(async () => {
      await waitForExpect(() => {
        wrapper.update();
        expect(wrapper.contains(<LoadingSpinner />)).toBeFalsy();
      });
    });
  };

  it('renders normally', async () => {
    const wrapper = getWrapper();

    await waitForContent(wrapper);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('handlePreview and handleCancelPreview should work as expected', async () => {
    const wrapper = getWrapper();

    await waitForContent(wrapper);
    expect(wrapper.find('CustomerMessageForm').prop('previewHtml')).toEqual(undefined);

    act(() => {
      const handlePreview = wrapper.find('CustomerMessageForm').prop('handlePreview') as Function;
      handlePreview('Tm90aWZpY2F0aW9uVGVtcGxhdGVOb2RlOjE=');
    });
    wrapper.update();
    expect(wrapper.find('CustomerMessageForm').prop('previewHtml')).toEqual(mockHtml);

    act(() => {
      const handleCancelPreview = wrapper.find('CustomerMessageForm').prop('handleCancelPreview') as Function;
      handleCancelPreview();
    });
    wrapper.update();
    expect(wrapper.find('CustomerMessageForm').prop('previewHtml')).toEqual(undefined);
  });

  it('handlePreview should set "previewHtml" as undefined if template preview does not exist', async () => {
    const wrapper = getWrapper();

    await waitForContent(wrapper);
    expect(wrapper.find('CustomerMessageForm').prop('previewHtml')).toEqual(undefined);

    act(() => {
      const handlePreview = wrapper.find('CustomerMessageForm').prop('handlePreview') as Function;
      handlePreview('SPAM-AD');
    });
    wrapper.update();
    expect(wrapper.find('CustomerMessageForm').prop('previewHtml')).toEqual(undefined);
  });
});
