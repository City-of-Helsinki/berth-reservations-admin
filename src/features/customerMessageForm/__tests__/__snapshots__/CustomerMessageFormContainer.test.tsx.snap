// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerMessageFormContainer renders normally 1`] = `
<form
  class="form"
>
  <h4
    class="text brand h4"
  >
    VIESTI
  </h4>
  <div
    class="grid cols2"
  >
    <div
      class="Select-module_root__Ka5uO select"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="templateId-toggle-button"
        id="templateId-label"
      >
        Viestipohja
        <span
          aria-hidden="true"
          class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
        >
          *
        </span>
      </label>
      <div
        class="Select-module_wrapper__1WQXs"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="templateId-label templateId-toggle-button"
          aria-owns="templateId-menu"
          class="Select-module_button__1aIsm"
          id="templateId-toggle-button"
          type="button"
        >
          berth_application_created
          <svg
            aria-hidden="true"
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Select-module_angleIcon__2-9AD"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
                fill="currentColor"
              />
            </g>
          </svg>
        </button>
        <ul
          aria-labelledby="templateId-label"
          aria-required="true"
          class="Select-module_menu__1H2aU"
          id="templateId-menu"
          role="listbox"
          style="max-height: 260px;"
          tabindex="-1"
        />
      </div>
    </div>
    <div
      class="recipientCount"
    >
      2 vastaanottajaa
    </div>
  </div>
  <div
    class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
  >
    <label
      class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
      for="subject"
    >
      Viestin aihe
      <span
        aria-hidden="true"
        class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
      >
        *
      </span>
    </label>
    <div
      class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
    >
      <input
        class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
        id="subject"
        required=""
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq message"
  >
    <label
      class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
      for="message"
    >
      Viesti
    </label>
    <div
      class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
    >
      <textarea
        class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
        id="message"
      />
    </div>
    <div
      class="TextInput-module_helperText__2dLR6 text-input_hds-text-input__helper-text__3V2KM"
    >
      Voit tarkastella viestiä esikatselussa ennen lähetystä.
    </div>
  </div>
  <div
    class="formActionButtons"
  >
    <button
      class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
      type="button"
    >
      <span
        class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
      >
        Peruuta
      </span>
    </button>
    <div
      class="formActionButtonsRight"
    >
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        type="button"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Esikatsele
        </span>
      </button>
      <button
        class="Button-module_button__3wgee button_hds-button__2A0je Button-module_primary__3bo_T button_hds-button--primary__2NVvO Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
        type="submit"
      >
        <span
          class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
        >
          Lähetä
        </span>
      </button>
    </div>
  </div>
</form>
`;
